### DISCUSSION TITLE
Is there an implementation of <form> for React Native?

### DISCUSSION THREAD

**Discussion by giuseppeg** [necolas/react-native-web#2056] [Q&A]
Anybody aware of any cross platform component that allows to use web forms everywhere? I am looking for a reimplementation of `<form>` for React Native.

**Comment by intergalacticspacehighway** [comment]
I think exporting a simple `<form/>` wrapper in .web.js and a no-op component in native.js should work. 
Form on web can accept `onSubmit` but native needs a `onPress` on submit button. 

I think the purpose of using a form component would be to achieve the form accessibility as described here, right?
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Form_Role

**Reply by intergalacticspacehighway** [reply]
Also, maybe someone can build a `Form` which accepts onSubmit and a `SubmitButton` component which uses React context and triggers it on press on native. 😅

**Comment by jottenlips** [comment]
I ended up making my own called `<Form/> ` that adds an` <input/> `for submitting on enter when rendered to web. It will also add `<form/> ` element to the HTML.
